pages/api/address/transactions/[address].ts

```diff
@@ -32,18 +33,22 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
       if (req.query.address === '' || req.query.address === undefined) {
         throw new Error(ADDRESS_NOT_PROVIDED_400.message)
       }
+      let transactions = []
       const address = parseAddress(req.query.address as string)
-      const transactions = await fetchAddressTransactions(address)
+
+      const addressInDB = (await fetchAddressesInList([address])).length
```

Comment:
Since it's only one address, makes more sense to use `fetchAddressBySubstring` than `fetchAddressesInList`. Also, you should not include the `.length` on the variable declaration so that the name `addressInDB` reflects better what the variable is and also mainly because then you don't have to always call `fetchAddressTransactions` down there, since `addressInDB.transactions` will return you exactly that for the case that the address exists.

Reply:
Will do!